edb/schema/objects.py

```diff
@@ -505,30 +518,79 @@ def __new__(
                 f' as {mcls._all_types[name]!r}'
             )
 
-        for k, v in tuple(clsdict.items()):
-            if isinstance(v, RefDict):
-                mydicts[k] = v
+        for k, field in tuple(clsdict.items()):
+            if isinstance(field, RefDict):
+                mydicts[k] = field
                 continue
-            if not isinstance(v, struct.ProtoField):
+            if not isinstance(field, struct.ProtoField):
                 continue
-            if not isinstance(v, Field):
+            if not isinstance(field, Field):
                 raise TypeError(
                     f'cannot create {name} class: schema.objects.Field '
-                    f'expected, got {type(v)}')
+                    f'expected, got {type(field)}')
 
-            v.name = k
-            myfields[k] = v
+            field.name = k
+            myfields[k] = field
             del clsdict[k]
 
-            if v.is_schema_field:
-                getter_name = f'get_{v.name}'
+            if isinstance(field, SchemaField):
+                getter_name = f'get_{field.name}'
                 if getter_name in clsdict:
                     # The getter was defined explicitly, move on.
                     continue
-                clsdict[getter_name] = (
-                    lambda self, schema, *, _fn=v.name:
-                        self._get_schema_field_value(schema, _fn)
-                )
+
+                ftype = field.type
+                if issubclass(ftype, s_abc.Reducible):
+                    def reducible_getter(
```

Comment:
I'd add a comment that you care about perf so these are closures and not toplevel funcs with functools.partial wrapping them.